### PR TITLE
Quick backport of scale mode feature from multires

### PR DESCRIPTION
**Pull request by ds-sloth**
This PR simply contains the screen scale mode from the multires branch for the main branch. I am trying to simplify the future multires review process by splitting smaller features into their own PRs.

The feature is tested from the multires branch and fully ready for general use. Fine to include in 1.3.6.1.

I quickly added some small additional quality-of-life changes: the scaling mode prompt and string are localizable, and the window doesn't try to resize when fullscreen or maximized.

Scr_2023-08-12_20-41-40

### CODE REVIEW

File: src/change_res.cpp [hunk 1]
```diff
@@ -19,21 +19,27 @@
...
-
 void SetOrigRes()
 {
     XWindow::setFullScreen(false);
     resChanged = false;
 
 #ifndef __EMSCRIPTEN__
-    XWindow::setWindowSize(ScreenW, ScreenH);
+    if(g_videoSettings.scaleMode == SCALE_FIXED_05X)
+        XWindow::setWindowSize(ScreenW/2, ScreenH/2);
```
Comment (Wohlstand) [inline src/change_res.cpp:38]: Please add spaces between operators and operands like `ScreenW / 2`.

File: src/change_res.cpp [hunk 2]
```diff
@@ -19,21 +19,27 @@
...
+    else if(g_videoSettings.scaleMode == SCALE_FIXED_2X)
+        XWindow::setWindowSize(ScreenW*2, ScreenH*2);
```
Comment (Wohlstand) [inline src/change_res.cpp:40]: Please add spaces between operators and operands like `ScreenW / 2`.

File: src/core/sdl/render_sdl.cpp [hunk 3]
```diff
@@ -18,6 +18,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <SDL2/SDL.h>
```
Comment (Wohlstand) [inline src/core/sdl/render_sdl.cpp:21]: What the reason to include the main SDL header? Does it contain anything that other smaler headers don't have? (for example, `SDL_Init()` or `SDL_Quit()`). But, I guess, `SDL_Hint` also here, but I don't remind exactly...

**Review by Wohlstand — CHANGES_REQUESTED** [review]
Overall looks fine. However, I have one code style quibble, and one small question.

File: src/core/sdl/render_sdl.cpp (hunk 3 above)
Comment (ds-sloth) [reply src/core/sdl/render_sdl.cpp]: Ah, SDL_hint, thank you.

**Comment by Wohlstand** [comment]
Okay, I fixed these things by myself, I'll merge.

**Pull request merged**